libc/src/stdio/linux/fdopen.cpp

```diff
@@ -0,0 +1,76 @@
+//===-- Implementation of fprintf -------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/stdio/fdopen.h"
+
+#include "include/llvm-libc-macros/generic-error-number-macros.h"
```

Comment:
use the proxy header (`hdr/errno_macros.h`) instead of directly including the public header. Same for the fcntl macros.